galsim/wfirst/wfirst_backgrounds.py

```diff
@@ -0,0 +1,175 @@
+    ecliptic_lon = abs(ecliptic_lon)
+    # And latitude symmetries:
+    if ecliptic_lat < 0.:
+        ecliptic_lat = abs(ecliptic_lat)
+    sin_ecliptic_lat = np.sin(np.pi*ecliptic_lat/180.)
+
+    # Take the lookup table, and turn negative numbers (indicating failure because of proximity to
+    # sun) to large positive values so that we can identify them as bad after interpolation.
+    max_sky = np.max(bandpass._sky_level)
+    sky_level = bandpass._sky_level.copy()
+    sky_level[sky_level<0] = 1.e6
+
+    # Interpolate!  We could do a full 2d interpolation, but this seems a bit like overkill.  We're
+    # going to first interpolate in latitude, then in longitude.
+    # Equal spacing in sin(latitude).
+    all_sin_lat_vals = np.sin(np.pi*bandpass._ecliptic_lat/180.)
+    sin_lat_vals = np.sort(np.array(list(set(all_sin_lat_vals))))
+    n_sl = len(sin_lat_vals)
+    d_sl = sin_lat_vals[1]-sin_lat_vals[0]
+    # Handle edge cases:
+    if sin_ecliptic_lat <= sin_lat_vals[0]:
+        sky_table_1d = sky_level[all_sin_lat_vals == sin_lat_vals[0]]
+    elif sin_ecliptic_lat >= sin_lat_vals[n_sl-1]:
+        sky_table_1d = sky_level[all_sin_lat_vals == sin_lat_vals[n_sl-1]]
+    else:
+        interp_val = (sin_ecliptic_lat - sin_lat_vals[0])/d_sl
+        lower_ind = int(np.floor(interp_val))
+        frac = interp_val - lower_ind
+        sky_table_1d = (1.-frac)*sky_level[all_sin_lat_vals == sin_lat_vals[lower_ind]] \
+            + frac*sky_level[all_sin_lat_vals == sin_lat_vals[lower_ind+1]]
+    # Now we have a table to interpolate on the set of galactic longitude values.
+    long_vals = np.sort(np.array(list(set(bandpass._ecliptic_lon))))
+    n_l = len(long_vals)
+    d_l = long_vals[1]-long_vals[0]
+    # Handle edge cases:
+    if ecliptic_lon <= long_vals[0]:
+        sky_val = sky_table_1d[0]
+    elif ecliptic_lon >= long_vals[n_l-1]:
+        sky_val = sky_table_1d[n_l-1]
+    else:
+        interp_val = (ecliptic_lon - long_vals[0])/d_l
+        lower_ind = int(np.floor(interp_val))
+        frac = interp_val - lower_ind
+        sky_val = (1.-frac)*sky_table_1d[lower_ind] + frac*sky_table_1d[lower_ind]
+
+    # If the result is too large, then raise an exception: we should not look at this position!
+    if sky_val > max_sky:
+        raise ValueError("Position is too close to sun!  Would not observe here.")
+
+    # Now, convert to the right units, and return.  (See docstring for explanation.)
+    # First, multiply by the effective collecting area in m^2.
+    eff_area = 0.25 * np.pi * galsim.wfirst.diameter**2 * (1. - galsim.wfirst.obscuration**2)
+    sky_val *= eff_area
+    # Then multiply by pixel area in arcsec^2.
+    sky_val *= galsim.wfirst.pixel_scale**2
+    # Optionally multiply by exposure time (if it was given).
+    if exp_time is not None:
+        sky_val *= exp_time
+
+    # The result is now in e-/pix (if exposure time was given), or e-/pix/s (if exposure time was
+    # not given).
```

Comment:
The routine `BaseWCS.makeSkyImage()` takes the sky level in ADU/arcsec*2.  So I think it would be better to conform to that convention here as well.  i.e. don't apply the pixel scale, but do apply the gain.